ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/services/ClaimMigrator.java

```diff
@@ -176,141 +216,135 @@ private void createCase(
-                        .getMadeBy() == MadeBy.CLAIMANT)
-                        && claim.getClaimantResponse().isPresent());
-                case AGREEMENT_COUNTER_SIGNED_BY_DEFENDANT:
-                    return claim.getSettlement().isPresent()
-                        && (claim.getSettlement().get().getLastStatement().getType() == StatementType.COUNTERSIGNATURE)
-                        && claim.getClaimantResponse().isPresent();
-                case AGREEMENT_REJECTED_BY_DEFENDANT:
-                    return claim.getSettlement().isPresent()
-                        && claim.getSettlement().get().getLastStatement().getType() == StatementType.REJECTION
-                        && claim.getSettlement().get().getLastStatement().getMadeBy() == MadeBy.DEFENDANT
-                        && claim.getClaimantResponse().isPresent();
-                case INTERLOCATORY_JUDGEMENT:
-                    return claim.getClaimantResponse().isPresent()
-                        && claim.getClaimantResponse().get().getType() == ClaimantResponseType.ACCEPTATION
-                        && claim.getResponse().isPresent()
-                        && !PartyUtils.isCompanyOrOrganisation(claim.getResponse().get().getDefendant())
-                        && ((ResponseAcceptation) claim.getClaimantResponse().get()).getFormaliseOption().isPresent()
-                        && ((ResponseAcceptation) claim.getClaimantResponse().get()).getFormaliseOption()
-                        .get() == FormaliseOption.REFER_TO_JUDGE;
-
-                case REJECT_ORGANISATION_PAYMENT_PLAN:
-                    return claim.getClaimantResponse().isPresent()
-                        && claim.getClaimantResponse().get().getType() == ClaimantResponseType.ACCEPTATION
-                        && claim.getResponse().isPresent()
-                        && PartyUtils.isCompanyOrOrganisation(claim.getResponse().get().getDefendant())
-                        && ((ResponseAcceptation) claim.getClaimantResponse().get()).getFormaliseOption().isPresent()
-                        && ((ResponseAcceptation) claim.getClaimantResponse().get()).getFormaliseOption()
-                        .get() == FormaliseOption.REFER_TO_JUDGE;
-                case REFER_TO_JUDGE_BY_CLAIMANT:
-                    return claim.getReDeterminationRequestedAt().isPresent()
-                        && claim.getReDetermination().isPresent()
-                        && claim.getReDetermination().get().getPartyType() == MadeBy.CLAIMANT;
-                case REFER_TO_JUDGE_BY_DEFENDANT:
-                    return claim.getReDeterminationRequestedAt().isPresent()
-                        && claim.getReDetermination().isPresent()
-                        && claim.getReDetermination().get().getPartyType() == MadeBy.DEFENDANT;
-                default:
-                    return false;
-            }
+        switch (event) {
```

Comment:
maybe we should include some try/catch handling with logging here in case of any unknown exceptions, NPEs - or will it be handled higher up?

Reply:
yes it is handled higher up